test/dom/Value.ts

```diff
@@ -1,161 +1,179 @@
                 new Date("1970-01-01T00:00:00Z"),
-                ['foo'],
                 IonTypes.TIMESTAMP,
-                null,
                 (jsValue, domValue) => domValue.dateValue().getTime() === jsValue.getTime()
             )
         );
-        it('Timestamp',
+        describe('Timestamp',
             domValueTest(
                 Timestamp.parse("1970-01-01T00:00:00Z"),
-                ['foo'],
                 IonTypes.TIMESTAMP,
-                null,
                 (jsValue, domValue) => domValue.timestampValue().equals(jsValue)
             )
         );
-        it('string',
+        describe('string',
             domValueTest(
                 'Hello',
-                ['foo'],
                 IonTypes.STRING,
             )
         );
-        it('String',
+        describe('String',
             domValueTest(
                 'Hello',
-                ['foo'],
                 IonTypes.STRING,
-                String
+                jsEqualsIon,
+                new String('Hello')
             )
         );
-
-        const blobData = new Uint8Array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]);
-        it('Blob',
+        describe('Blob',
             domValueTest(
-                blobData,
-                ['foo'],
+                new Uint8Array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]),
                 IonTypes.BLOB,
-                null,
                 (jsValue, domValue) =>
-                    domValue.uInt8ArrayValue().every(
-                        (byte, index) => byte === blobData[index]
+                    jsValue.every(
+                        (byte, index) => byte === domValue[index]
                     )
             )
         );
-
-        const listData = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
-        it('List',
+        describe('List',
             domValueTest(
-                listData,
-                ['foo'],
+                [0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
                 IonTypes.LIST,
-                null,
                 (jsValue, domValue) =>
-                    domValue.every(
-                        (ionValue, index) => ionValue.numberValue() == jsValue[index]
+                    jsValue.every(
+                        (listItem, index) => listItem === domValue[index].numberValue()
                     )
                 )
         );
-
-        const structData = {foo: 7, bar: true, baz: 98.6, qux: 'Hello'};
-        it('Struct',
+        describe('Struct',
             domValueTest(
-                structData,
-                ['foo'],
+                {foo: 7, bar: true, baz: 98.6, qux: 'Hello'},
                 IonTypes.STRUCT,
-                null,
                 (jsValue, domValue) =>
                     Object.entries(jsValue)
                           .every(([key, value]) => domValue.get(key).valueOf() === value)
             )
         );
+        describe('List inside Struct',
+            domValueTest(
+                {foo: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]},
```

Comment:
fwiw, i'd probably reverse (or shuffle) the list of numbers to make sure the test isn't asserting against list index instead of value

same for the 'List' test above

Reply:
Reversed.